src/openforms/authentication/api/views.py

```diff
@@ -62,3 +70,38 @@ def delete(self, request, *args, **kwargs):
 
         request.session.flush()
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class SubmissionLogoutView(GenericAPIView):
+    authentication_classes = ()
+    permission_classes = (ActiveSubmissionPermission,)
+    serializer_class = (
+        serializers.Serializer
+    )  # just to shut up some warnings in drf-spectacular
+
+    queryset = Submission.objects.all()
+    lookup_field = "uuid"
+
+    @extend_schema(
+        operation_id="submission_session_destroy",
+        summary=_("Delete session"),
+        description=_(
+            "Calling this endpoint will clear the current form and submission from the session."
+        ),
+    )
+    @transaction.atomic()
+    def delete(self, request, *args, **kwargs):
+        submission = self.get_object()
+
+        remove_submission_from_session(submission, request.session)
+
+        if FORM_AUTH_SESSION_KEY in request.session:
+            del request.session[FORM_AUTH_SESSION_KEY]
+
+        if not submission.auth_attributes_hashed:
+            submission.hash_identifying_attributes(save=True)
+
+        for plugin in register.iter_enabled_plugins():
+            plugin.logout(request)
```

Comment:
```suggestion
        for plugin in register:
            plugin.logout(request)
```
isn't the default behaviour to only loop over enabled plugins? so that could clean up the code a bit without leaking implementation details maybe?

Reply:
I think things could go wrong if we iterate over all plugins and try to logut on one that is not configured (eg: missing urls/config etc)

Reply:
I checked after discussion and the default register iterator iterates over everything included disabled plugins so we definitely want .iter_enabled_plugins() here.